services/web-app/utils/use-query-state.js

```diff
@@ -75,19 +66,21 @@ export const useQueryState = (
 
       // Don't rely on router.query here because that would cause unnecessary renders when other
       // query parameters change
-      const query = new URLSearchParams(window.location.search)
+      const query = queryString.parseUrl(window.location.search, {
+        ...queryStringConfig,
+        parseNumbers,
+        parseBooleans
+      }).query
 
-      if (newValue === null || newValue === undefined) {
-        query.delete(key)
-      } else {
-        query.set(key, serialize(newValue))
-      }
+      query[key] = newValue
 
-      router.replace(`?${query.toString()}`, undefined, { shallow: true, scroll: false })
+      // Change query state without rerendering page
+      history.replaceState(null, null, `?${queryString.stringify(query, queryStringConfig)}`)
```

Comment:
Does this mean that no matter how many times you update the query string, you'll only ever have one history entry? If so, is that desirable? A lot of catalog-type websites I use don't behave that way; they update the URL each time the filter is applied (sometimes with debouncing or explicit "apply" buttons).

Reply:
the update function only runs on a per-key basis so only one query string key should be updated at a time

Reply:
That makes sense, but I was wondering (user-experience-wise) if a particular key is updated 3 or 4 times in a row, what would the user's browser history look like? Would they see a history entry for each filter update? Or would they see only a single history entry? I'd argue that they should see one history entry per key update so they can use their back button to traverse how they'd changed their filters over time.

Reply:
did a test with 
`
   useEffect(() => {
     setQuery('q1')
     setQuery('q2')
     setQuery('q3')
     setQuery('q4')
   }, [sort])
`
and tried checking `history.back()` and confirmed can navigate back to all the changed filters (`?q=3`, `?q=2`, `?q=1`)